aspnet/mvc/models/formatting.rst

```diff
@@ -1,8 +1,223 @@
+
+Browsers and Content Negotiation
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Unlike typical API clients, web browsers tend to supply ``Accept`` headers that include a wide array of formats, including wildcards. By default, when the framework detects that the request is coming from a browser, it will ignore the ``Accept`` header and instead return the content in the application's configured default format (JSON unless otherwise configured). This provides a more consistent experience when using different browsers to consume APIs.
+
+If you would prefer your application honor browser accept headers, you can configure this as part of MVC's configuration by setting ``RespectBrowserAcceptHeader`` to ``true`` in the ``ConfigureServices`` method in *Startup.cs*.
+
+.. code-block:: c#
+
+  services.AddMvc(options =>
+  {
+    options.RespectBrowserAcceptHeader = true; // false by default
+  }
+
+
+Configuring Formatters
+----------------------
+
+If your application needs to support additional formats beyond the default of JSON, you can add these as additional dependencies in *project.json* and configure MVC to support them. There are separate formatters for input and output. Input formatters are used by :doc:`model-binding`; output formatters are used to format API responses. You can also configure :doc:`custom-formatters`.
+
+Configuring the JSON Formatter
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+A common request when formatting responses using JSON is to use JavaScript-style "camel" casing (the first character of each property is lowercase, then uppercase is used to begin each additional word making up the name). Support for camel casing can be added by using the ``AddJsonOptions`` extension method when configuring MVC:
+
+.. code-block:: c#
+
+  services.AddMvc()
+    .AddJsonOptions(o =>
+      o.SerializerSettings.ContractResolver = 
+        new CamelCasePropertyNamesContractResolver());
+
+Replace the default ``SerializerSettings.ContractResolver`` with a new instance of ``CamelCasePropertyNamesContractResolver``. Before this change, an object with a property of "FirstName" and a value of "Steve" would be serialized to JSON as ``{"FirstName":"Steve"}``; after the change to use camel case, it would serialize as ``{"firstName":"Steve"}``.
+
+Adding XML Format Support
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To add support for XML formatting, add the "Microsoft.AspNetCore.Mvc.Formatters.Xml" package to your *project.json*'s list of dependencies.
+
+Add the XmlSerializerFormatters to MVC's configuration in *Startup.cs*:
```

Comment:
You really have two options here as there are two sets of XML formatters. We have separate XML formatters that use XmlSerializer and that use the DataContractSerializer. Both should be mentioned.